src/RedialContext.js

```diff
@@ -165,36 +174,60 @@ export default class RedialContext extends Component {
       aborted,
       abort,
       loading: true,
+      beforeTransitionCompleted: false,
       prevRenderProps: this.state.aborted() ? this.state.prevRenderProps : this.props.renderProps,
     });
 
-    const promises = [this.runBlocking(
-      this.props.blocking,
-      components,
-      renderProps,
-      force,
-      bail
-    )];
-
     if (this.props.parallel) {
-      promises.push(this.runDeferred(
-        this.props.defer,
+      this.runAfterTransition(
+        this.props.afterTransition,
         components,
         renderProps,
         force,
         bail
-      ));
+      )
+      .then(() => {
+        if (this.state.beforeTransitionCompleted) {
+          this.props.onCompleted('afterTransition');
+        }
```

Comment:
If we are in `parallel`-mode, won't we have a risk of calling `onCompleted` twice with `'afterTransition'`? Here, and line L280